### PR TITLE
Feature/exclusion of elements

### LINKED ISSUE #6: Create the ability to exclude data elements from Serialization/Deserialization

### LINKED ISSUE #16: Add the ability to flag a property to be excluded on deserialization

### PR DESCRIPTION
**Pull request by brettminnie**
@eddiejaoude when you get a moment

Will close #16 and close #6

### CODE REVIEW

File: src/BreakfastSerializer/ConfigurableProperty.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    /**
+     * @var array
+     */
+    protected static $configurationData;
+
+    /**
+     * @param string $pathName
+     * @return IsConfigurable
+     */
+    public function setConfigurationPath($pathName = './config')
```
Comment (eddiejaoude) [inline src/BreakfastSerializer/ConfigurableProperty.php]: Would this be better in a default config file?

Comment (brettminnie) [inline src/BreakfastSerializer/ConfigurableProperty.php]: Possibly, though I wanted it to run without dicking about with a bootstrap, should probably include the default directory as well...

File: composer.json [hunk 2]
```diff
@@ -11,7 +11,8 @@
         }
     ],
     "require": {
-        "php": ">=5.3.2"
+        "php": ">=5.3.2",
```
Comment (eddiejaoude) [inline composer.json]: You are only running travis tests for `v5.5` & `v5.6`.

File: src/BreakfastSerializer/Serializer.php [hunk 3]
```diff
@@ -153,4 +159,24 @@ protected function cleanVariableName($variableName, $containingClass)
...
     }
 
+    /**
+     * @param string $propertyName
+     * @param string $className
+     * @return bool
+     */
+    public function isExcluded($propertyName, $className)
+    {
+        if (true === isset($this->getConfiguration()['exclusions'][$className]['excludeVariables'])) {
```
Comment (eddiejaoude) [inline src/BreakfastSerializer/Serializer.php]: In your `composer.json` file you have `"php": ">=5.3.2"`, is this not `5.4` and above?

File: src/BreakfastSerializer/ConfigurableProperty.php (hunk 1 above)
Comment (eddiejaoude) [inline src/BreakfastSerializer/ConfigurableProperty.php]: Ah ok. Just a thought :smile:

Comment (brettminnie) [inline src/BreakfastSerializer/ConfigurableProperty.php]: I will implement default config and one or two other tweaks in a later pr, it's a good idea, I just need to figure out how to do it neatly

File: src/BreakfastSerializer/Serializer.php (hunk 3 above)
Comment (brettminnie) [inline src/BreakfastSerializer/Serializer.php]: Nicely spotted

File: composer.json (hunk 2 above)
Comment (brettminnie) [inline composer.json]: Will amend

File: src/BreakfastSerializer/ConfigurableProperty.php (hunk 1 above)
Comment (brettminnie) [inline src/BreakfastSerializer/ConfigurableProperty.php]: Raised as #18

Comment (eddiejaoude) [inline src/BreakfastSerializer/ConfigurableProperty.php]: :+1:

File: composer.json (hunk 2 above)
Comment (brettminnie) [inline composer.json]: @eddiejaoude updated

**Pull request merged**